pkg/kubefed2/federate/util.go

```diff
@@ -49,3 +58,104 @@ func SetBasicMetaFields(resource *unstructured.Unstructured, apiResource metav1.
 		resource.SetNamespace(namespace)
 	}
 }
+
+func getAPIResourceList(config *rest.Config, skipAPIResourceNames string) (map[string]metav1.APIResource, error) {
+	apiResourceLists, err := enable.GetServerPreferredResources(config)
+	if err != nil {
+		return nil, err
+	}
+
+	apiResources := make(map[string]metav1.APIResource)
+	for _, apiResourceList := range apiResourceLists {
+		if len(apiResourceList.APIResources) == 0 {
+			continue
+		}
+
+		gv, err := schema.ParseGroupVersion(apiResourceList.GroupVersion)
+		if err != nil {
+			return nil, errors.Wrap(err, "Error parsing GroupVersion")
+		}
+
+		group := gv.Group
+		for _, apiResource := range apiResourceList.APIResources {
+			if !apiResource.Namespaced || isAlreadyFederated(apiResource) ||
+				apiResourceMatchesSkipNames(apiResource, skipAPIResourceNames, group) {
+				continue
+			}
+
+			// TODO(irfanurrehman): Define a strategy to federate the latest available version
```

Comment:
I would like to see this strategy implemented before this PR merges to avoid throwing away the care that  `enable` takes to avoid enabling types for ambiguous input.

Reply:
Updated!